crates/services/p2p/src/p2p_service.rs

```diff
@@ -293,14 +301,20 @@ impl FuelP2PService {
             network_codec: codec,
             outbound_requests_table: HashMap::default(),
             inbound_requests_table: HashMap::default(),
+            gossipsub_peer_limiter: SizedHashset::new(
+                config.max_gossipsub_peers_connected as usize,
+            ),
+            request_response_peer_limiter: SizedHashset::new(
+                config.max_request_response_peers_connected as usize,
+            ),
```

Comment:
You probably already remember what I think of `as` conversions in production code 🫠

```suggestion
            gossipsub_peer_limiter: SizedHashset::new(
                usize::try_from(config.max_gossipsub_peers_connected)?,
            ),
            request_response_peer_limiter: SizedHashset::new(
                usize::try_from(config.max_request_response_peers_connected)?,
            ),
```